client/extensions/woocommerce/app/product-categories/form.js

```diff
@@ -28,12 +34,139 @@ class ProductCategoryForm extends Component {
 		editProductCategory: PropTypes.func.isRequired,
 	};
 
+	constructor( props ) {
+		super( props );
+
+		const { category } = props;
+		const image = ( category && category.image ) || {};
+
+		this.state = {
+			id: image.id || null,
+			src: image.src || null,
+			placeholder: null,
+			transientId: null,
+			isUploading: false,
+		};
+
+		this.debouncedSetDescription = debounce( this.setDescription, 200 );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.category !== this.props.category ) {
+			const image = ( nextProps.category && nextProps.category.image ) || {};
+			this.setState( {
+				id: image.id || null,
+				src: image.src || null,
+			} );
+		}
+	}
+
 	setName = e => {
 		const { siteId, category, editProductCategory } = this.props;
 		const name = e.target.value;
 		editProductCategory( siteId, category, { name } );
 	};
 
+	setDescription = description => {
+		const { siteId, category, editProductCategory } = this.props;
+		editProductCategory( siteId, category, { description } );
+	};
+
+	onSelect = files => {
+		const file = head( files );
+		this.setState( {
+			placeholder: file.preview,
+			transientId: file.ID,
+			isUploading: true,
+		} );
+	};
+
+	onUpload = file => {
+		const { siteId, editProductCategory, category } = this.props;
+		const image = {
+			src: file.URL,
+			id: file.ID,
+		};
+		this.setState( {
+			...image,
+			transientId: null,
+			isUploading: false,
+		} );
+		editProductCategory( siteId, category, { image } );
+	};
+
+	onError = () => {
+		this.setState( {
+			placeholder: null,
+			transientId: null,
+			isUploading: false,
+		} );
+	};
+
+	removeImage = () => {
+		const { siteId, editProductCategory, category } = this.props;
+		this.setState( {
+			placeholder: null,
+			transientId: null,
+			isUploading: false,
+			src: null,
+			id: null,
+		} );
+		editProductCategory( siteId, category, { image: {} } );
+	};
+
+	renderImage = () => {
+		const { src, placeholder, isUploading } = this.state;
+
+		let image = null;
+		if ( src && ! isUploading ) {
+			image = (
+				<figure>
+					<ImagePreloader
+						src={ src }
+						placeholder={ ( placeholder && <img src={ placeholder } /> ) || <span /> }
+					/>
+				</figure>
+			);
+		} else if ( isUploading ) {
+			image = (
+				<figure>
+					<img src={ placeholder || <span /> } />
```

Comment:
A `<span />` as the img src?

Reply:
Doh, thanks. A span is used as a placeholder for `<ImagePreloader>` a few lines up, but yeah -- we shouldn't have that here. Removed in https://github.com/Automattic/wp-calypso/pull/21347/commits/f70504b79f35f0e1f58c2cf0021aff01f381dd98.